### ISSUE TITLE
Registering custom languages

### ISSUE THREAD

**Issue by David-Moreira** [serdarciplak/BlazorMonaco#26]
Hello @serdarciplak ,
I'm curious if you're still supporting this library, seems like no new contributions since December. 
I'm actually quite interested, in using this one in a big project, and it is important to me, if there's activity/support, otherwise I will probably have to respectfully clone it if issues come up.

**Comment by serdarciplak** [comment]
Hi @David-Moreira
I'm actively maintaining the repo. But if you're asking for 7/24 instant support, I cannot guarantee that as you may guess.
Due to the nature of open source, you can fork any repo anytime. If I were you, I would start using this repo. If I need changes and I cannot get responses as I need, I would go fork it then. But that's all your choice. Take care :)

**Comment by David-Moreira** [comment]
Thanks @serdarciplak
Personally I already needed some features like setting a new language, monarch provider, completion tokens, markers... If you'd like to eventually add to your project. 
But this is already a nice starting point. 
Alot of "plumbing" already done and that's awesome.

**Comment by serdarciplak** [comment]
Some features you need may already be implemented. I've tried to keep everything in parallel to the original Monaco Editor javascript API. So, you may check how to do something in the original way, and then check out if BlazorMonaco supports it. Please let me know if you still need something to be implemented and I'll check it out.

For example, you already can set a new language on BlazorMonaco.

The original API way : 
```javascript
var model = editor.getModel();
monaco.editor.setModelLanguage(model, 'csharp');
```

BlazorMonaco way : 
```csharp
var model = await editor.GetModel();
await MonacoEditor.SetModelLanguage(model, "csharp");
```

**Comment by David-Moreira** [comment]
Hello @serdarciplak 
Yea there's quite alot of stuff already implemented, thank you for your work.

What I meant by new language is registering a custom one.
I am not wrong to say that these are missing am I?

```
// Register a new language
monaco.languages.register({ id: 'mySpecialLanguage' });

// Register a completion item provider for the new language
monaco.languages.registerCompletionItemProvider('mySpecialLanguage', {

// Register a tokens provider for the language
monaco.languages.setMonarchTokensProvider('mySpecialLanguage', {
```

https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages

**Comment by serdarciplak** [comment]
You're right. You cannot register custom languages in Blazor yet. As of now, you can register the language in Javascript and use it in Blazor.

**Comment by sfmskywalker** [comment]
> BlazorMonaco way :
> 
> ```cs
> var model = await editor.GetModel();
> await MonacoEditor.SetModelLanguage(model, "csharp");
> ```

In case anyone else stumbles upon this: as of this writing, the correct way is to use the `Global` static class. Example:

```csharp
var model = await editor.GetModel();
await Global.SetModelLanguage(model, "csharp");
```

@serdarciplak Please do correct me if I'm wrong! 😄

**Comment by serdarciplak** [comment]
Yes, that's correct. The `SetModelLanguage` method is moved to the `BlazorMonaco.Editor.Global` class in a newer release.
